controllers/pullRequestsController.js

```diff
@@ -1,57 +1,82 @@
 const githubService = require('../services/githubService')
 
+/**
+ * Get Latest PRs in open or stale state for Real Dev Squad repos
+ *
+ * @param {boolean} page page number to retrieve
+ * @param {boolean} perPage number of PRs per page
+ * @param {boolean} isOpen boolean value indication open or stale
+ * @param {string} username retrieve PRs of username
+ *
+ * @returns {Array} List of PRs based on provided conditions
+ */
+const getPRs = async ({ page = 1, perPage = 10, isOpen = true, username = null }) => {
+  try {
+    const { data } = await githubService.fetchPRs({ page, perPage, isOpen, username })
+    if (!data.total_count) {
+      return []
+    }
+    return githubService.extractPRdetails(data)
+  } catch (err) {
+    logger.error(`Error while processing pull requests: ${err}`)
+    throw err
+  }
+}
+
 /**
  * Collects all pull requests and sends only required data for each pull request
  *
  * @param req {Object} - Express request object
  * @param res {Object} - Express response object
  */
-
 const getUserPRs = async (req, res) => {
   try {
-    const { data } = await githubService.fetchPRsByUser(req.params.username)
+    const { username } = req.params
+
+    let message
+    const prs = await getPRs({ username })
```

Comment:
can be called as `pullRequests` so it'd become better while returning (read comment below)